Experiments_toolkit.py

```diff
@@ -57,151 +50,143 @@ def splitting_up_data(dictionary, dict_key, train, percent_train, test, percent_
+def create_dataset(pos, neg, max_samples):
+    """
+       Description: Create a tf.data.Dataset composed of samples from the pos and neg list 
+
+       Args:
+           pos -- np array (Numpy arrays of positive windows, [(sensor_dims), num_samples])
+           neg -- np arrya (Numpy arrays of negative windows, [(sensor_dims), num_samples])
+           max_size -- int (number of samples to be taken from pos and neg to
+                                                be used in the dataset. If max_size < 0, then all 
+                                                samples from both pos and neg will be used) 
+
+       Return:
+           dataset -- tf.data.Dataset (TensorFlow dataset composed of the 
+                                  positive and negative lists of data passed into the 
+                                  function with the corresponding labels)
+     """
+    np.random.shuffle(pos)
+    np.random.shuffle(neg)
+    if (max_samples > 0):
+        pos = pos[:max_samples]
+        neg = neg[:max_samples]
+    examples = np.concatenate((pos, neg))
+    pos_label = np.ones(len(pos))
+    neg_label = np.zeros(len(neg))
+    labels = np.concatenate((pos_label, neg_label))
+    dataset = tfdata.Dataset.from_tensor_slices((examples, labels))
+    return dataset
+
+
+def preprocess_add_augmentation(data, augmentation_function, sigma):
     """
        Description: Iterate over DATA and proprocess the traces by adding an augmentation function
                                 to the data.
 
        Args:
-           DATA -- np array (Numpy array where the last row is time steps)
+           data -- list (List of numpy array where the last row is time steps, [(sensor_dims + 1), num_samples])
            dictionary -- dict (Dictionary to store augmented traces)
            pos_key -- string (Key in dictionary to store augmented windows) 
            augmentation_function -- function (Desired augmentation function of the form:
-                                                                                        augment(np_array data, int sigma))
+                                              augment(np_array data, int sigma))
            sigma -- double (Level of augment to be passed to the augmentation function )
            size -- int (Window size of traces to be stored in dictionary)
 
        Return:
            dictionary -- dict (Dictionary with augmented windows of size = size in key pos_key)
      """
-    for j in range(len(DATA)):
-        sample = DATA[j]
-        length = len(np_sample[0][0])
-        if length < size:
-            continue
-        window = same_window_size(sample[:-1], size, 10)
-        transpose = window.transpose(0, 2, 1)
-        for i in range(len(transpose)):
-            augmented = augmentation_function(transpose[i], sigma)
-            dictionary[pos_key].extend([augmented])
-
-    return dictionary
-
-
-def DA_Rotation_specific(X, angle_low, angle_high, axis):
```

Comment:
are the augmentation functions in a new file now? I don't think that's submitted yet?

Reply:
Yeah, so the augmentation functions are in the forked repo in my github